draft-ietf-ppm-dap.md

```diff
@@ -1264,30 +1265,37 @@ enum {
 } PrepareStepState;
 
 struct {
-  ReportID report_id;
   PrepareStepState prepare_step_state;
   select (PrepareStep.prepare_step_state) {
-    case continued: opaque prep_msg<0..2^32-1>; /* VDAF preparation message */
+    case continued: opaque prep_msg<0..2^32-1>; /* VDAF preparation message or share */
     case finished:  Empty;
     case failed:    ReportShareError;
   };
 } PrepareStep;
 
 struct {
-  PrepareStep prepare_steps<1..2^32-1>;
-} AggregationJobResp;
+  ReportID report_id;
+  PrepareStep init_prep_step;
+  PrepareStep stepped_prep_step;
+} ReportAggregationInitResp;
+
+struct {
+  ReportAggregationInitResp report_shares<1..2^32-1>;
+} AggregationJobInitResp;
 ~~~
 
-The message is a sequence of PrepareStep values, the order of which matches that
-of the ReportShare values in `AggregationJobInitReq.report_shares`. Each report
-that was marked as invalid is assigned the `PrepareStepState` `failed`.
-Otherwise, the `PrepareStep` is either marked as continued with the output
-`prep_msg`, or is marked as finished if the VDAF preparation process is finished
-for the report share. The Helper's aggregation job is now in round 0.
+The message is a sequence of `ReportAggregationInitResp` values, the order of
+which matches that of the ReportShare values in
+`AggregationJobInitReq.report_shares`. Each report that was marked as invalid is
+assigned the `PrepareStepState` `failed`; if the prepare step corresponding to
+`init_prep_step` fails, `stepped_prep_step` is unspecified. Otherwise, the
+`PrepareStep`s are either marked as continued with the output `prep_msg`, or are
+marked as finished if the VDAF preparation process is finished for the report
+share. The Helper's aggregation job is now in round 0.
```

Comment:
Shouldn't it be "round 1", since the Helper has processed the round 1 prep message?

Reply:
I think that it is clearer if we maintain that the `round` counter counts DAP rounds (i.e. network round-trips), rather than VDAF rounds, since the DAP round is the "unit of continuation" -- if something goes wrong in transit during an aggregation continuation attempt, we'll be restarting from the beginning of the previous DAP round, rather than from a VDAF round that is traversed partway through a DAP round.